### PR TITLE
New documentation setup

### PR DESCRIPTION
**Pull request by frangio**
This PR is the result of the work we've been doing to improve our documentation infrastructure, as a first step towards improving the content. :slightly_smiling_face: 

This branch is deployed at [openzeppelin-docs-next.netlify.com](http://openzeppelin-docs-next.netlify.com/).

This is still wired up to the `next` branches of `solidity-docgen` and `openzeppelin-docsite`. Perhaps we should merge those into `master` before merging, but I wanted to drop this PR now.

### CODE REVIEW

**Comment by nventuro** [comment]
I think I found a bug (no idea where it is in the code though):

[Escrow](https://openzeppelin-docs-next.netlify.com/api/payment/escrow.html#escrow)'s link to primary redirects to https://openzeppelin-docs-next.netlify.com/api/payment/escrow.html#Secondary.primary(), while it should be https://openzeppelin-docs-next.netlify.com/api/ownership.html#Secondary.primary() (i.e. we need to look up in which section the parent contract is located).

**Comment by frangio** [comment]
Good catch, yes. This is in `solidity-docgen`. I'll take a look at it tomorrow.

**Comment by nventuro** [comment]
What is it that causes the docsite to be built for each branch? And is there a way to reduce the number of GitHub checks to just one (i.e. 'the docsite deployment was successful')?

File: docs/learn-about-utilities.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+- `myNumber.sub(otherNumber)`
+- `myNumber.div(otherNumber)`
+- `myNumber.mul(otherNumber)`
+- `myNumber.mod(otherNumber)`
+
+Easy!
+
+## Payment
+
+Want to split some payments between multiple people? Maybe you have an app that sends 30% of art purchases to the original creator and 70% of the profits to the current owner; you can build that with [`PaymentSplitter`](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/payment/PaymentSplitter.sol)!.
```
Comment (nventuro) [inline docs/learn-about-utilities.md]: We don't want to have links to `master`, but also probably don't want to pin them to a specific release. ideas?

File: docs/learn-about-utilities.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+.toEthSignedMessageHash()
+.recover(signature)
+```
+
+- [MerkleProof.sol](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/cryptography/MerkleProof.sol) — provides `verify(...)` for verifying merkle proofs.
+
+
+## Introspection
+
+In solidity, it's frequently helpful to know whether or not a contract supports an interface you'd like to use. ERC165 is a standard that helps do runtime interface detection. OpenZeppelin provides some helpers, both for implementing ERC165 in your contracts and querying other contracts:
```
Comment (nventuro) [inline docs/learn-about-utilities.md]:
```suggestion
In Solidity, it's frequently helpful to know whether or not a contract supports an interface you'd like to use. ERC165 is a standard that helps do runtime interface detection. OpenZeppelin provides some helpers, both for implementing ERC165 in your contracts and querying other contracts:
```

**Comment by nventuro** [comment]
We may want to include some sort of spell-check to the docs directory (and comments?) as well.

File: docs/learn-about-utilities.md (hunk 1 above)
Comment (frangio) [reply docs/learn-about-utilities.md]: Good point. These links should probably be handled by the same thing that creates the links within the API Reference. This piece of the puzzle is not built yet...

Comment (nventuro) [reply docs/learn-about-utilities.md]: I'd use the latest release for now then.

**Comment by frangio** [comment]
> What  is it that causes the docsite to be built for each branch?

It's configured in the Netlify admin panel.

> And is there  a way to reduce the number of GitHub checks to just one (i.e. 'the  docsite deployment was successful')?

I just toggled something that I think will remove it. Not sure if we'll be able to see that in this PR though.

**Comment by frangio** [comment]
The links should be fixed now. But the code in solidity-docgen keeps getting uglier and uglier. :upside_down_face:

**Review by IAbouzied — COMMENTED** [review]
LGTM!

**Pull request merged**